docs/src/ZIndex Classes.doc.js

```diff
@@ -1,181 +1,660 @@
+
+const modalZindex = new ComposedZIndex([SheetZindex]);
+
+const ModalWithZIndex = <Layer zIndex={modalZindex}><Modal/></Layer>
 ~~~
-  "
-        />
-      </Box>
-      <Text>
-        However, this is an escape hatch that should only be used in cases like needing to work with
-        a third party library. For any other case, a better approach is to wrap the component that
-        needs a zIndex in a Box.
-      </Text>
-      <Box padding={2} color="lightGray" rounding={2}>
-        <Markdown
-          text="
+`}
+      />
+      <MainSection.Card
+        cardSize="md"
+        type="don't"
+        description={`
+Use unnecessarily high fixed z-index values
 ~~~jsx
 import { FixedZIndex } from 'gestalt';
+
+const HeaderZindex = new FixedZIndex(1);
+
+const MenuZindex =new FixedZIndex(1000);
+
+const SheetZindex = new FixedZIndex(9999);
+
+const modalZindex = new FixedZIndex(100000);
+
+const ModalWithZIndex = <Layer zIndex={modalZindex}><Modal/></Layer>
+~~~
+`}
+      />
+    </MainSection.Subsection>
+    <MainSection.Subsection columns={2}>
+      <MainSection.Card
+        cardSize="md"
+        type="do"
+        description={`
+Wrap non-Gestaltifiable HTML tags in Box and pass zIndex classes
+~~~jsx
+import { FixedZIndex } from 'gestalt';
+
 const fixedZindex = new FixedZIndex(1);
+
+const canvas = (
+    <Box zIndex={fixedZindex}>
+      <canvas
+        id="myCanvas"
+        width="200"
+        height="100"
+        style="border:1px solid #000000;"></canvas>
+    </Box>
+);
 ~~~
-  "
-        />
-        <Flex alignItems="center" gap={2}>
-          <Icon icon="cancel" accessibilityLabel="Cancel" />
-          <Markdown
-            text="
+`}
+      />
+      <MainSection.Card
+        cardSize="md"
+        type="don't"
+        description={`
+Extract z-index values from both zIndex classes to use in non-Gestaltifiable HTML Tags. See [z-index in non-Gestalt components](#z-index-in-non-Gestalt-components) to learn more.
 ~~~jsx
-const customButton = <CustomButton zIndex={fixedZindex.index()}/>;
+import { FixedZIndex } from 'gestalt';
+
+const fixedZindex = new FixedZIndex(1);
+
+const canvas = (
+  <canvas
+    z-index={fixedZindex.index()}
+    id="myCanvas"
+    width="200"
+    height="100"
+    style="border:1px solid #000000;"></canvas>
+);
 ~~~
-  "
-          />
-        </Flex>
-        <Flex alignItems="center" gap={2}>
-          <Icon icon="check" accessibilityLabel="Cancel" />
-          <Markdown
-            text="
+`}
+      />
+    </MainSection.Subsection>
+    <MainSection.Subsection columns={2}>
+      <MainSection.Card
+        cardSize="lg"
+        type="do"
+        description={`
+Use Layer instead of zIndex when a component needs to visually break out of its parent container, for example, in the case of [modals](/Modal), [sheets](/Sheet) and [tooltips](/Tooltip). See [z-Index in foundational components](#zIndex-in-foundational-components) and [ZIndex in Layer](#zIndex-in-Layer) to learn more.
```

Comment:
I think the Don't example here would be something like
```js
<Box zIndex={1000000}>
  <Modal />
<Box>
```

Reply:
DONE